test/e2e/model-definition.js

```diff
@@ -0,0 +1,46 @@
+var DataSource = require('../..').DataSource;
+var MongoObjId = require('mongodb').ObjectId;
+var ObjectId = require('../../lib/objectid');
+var should = require('should');
+
+describe('model definition', function() {
+  context('ObjectId definition', function() {
+    var Todo;
+
+    before(function setup() {
+      var ds = new DataSource();
+      Todo = ds.define('Todo', {
+        id: {type: 'ObjectId', id: true}
+      });
+    });
+
+    it('should allow 12 byte strings', function() {
+      var todo = new Todo({id: 'hello1hello1'});
+
+      todo.id.should.be.an.instanceof(ObjectId);
+      todo.id.should.be.an.instanceof(MongoObjId);
```

Comment:
>Also the tests in this file duplicates the tests in test/objectid.test.js. What's the point of that?

They are testing from outside in vs inside out (ie. functional vs unit). They look like dupes, but should be tested independently. The e2e tests use things like `new Todo({type: 'ObjectId', ...})` vs `new ObjectId()`. IMO, LoopBack mixes unit/functional tests together way too much and shouldn't be. The different aspects should be tested independently and in isolation.